test/androidTest/java/org/thoughtcrime/securesms/espressoTests/MarkMessageAsUnreadTest.java

```diff
@@ -0,0 +1,109 @@
+    public ActivityTestRule<ConversationListActivity> mActivityTestRule = new ActivityTestRule<>(ConversationListActivity.class);
+
+    @Test
+    public void checkMarkAsUnreadAvailableTest() {
+        ConversationActions.createNewConversation("123456");
+        ConversationActions.enableSignalForSMS();
+        ConversationActions.sendMessage("Sup?");
+        ConversationActions.longPressMessageAt(0);
+        Expectations.checkIsDisplayed(R.id.menu_context_mark_as_unread);
+    }
+
+    @Test
+    public void checkMarkingAsUnreadTest() {
+        ConversationActions.createNewConversation("123456");
+        ConversationActions.enableSignalForSMS();
+        ConversationActions.sendMessage("Sup?");
+        ConversationActions.longPressMessageAt(0);
+        ConversationActions.markAsUnread();
+        ConversationActions.goToConversation("+15555123456");
+        Expectations.checkIsShowingDescendantWithId(EspressoUtil.nthChildOf(withId(android.R.id.list), 0), R.id.read_reminder);
+    }
+
+    @Test
+    public void checkMarkingMultipleAsUnreadTest() {
+        ConversationActions.createNewConversation("123456");
+        ConversationActions.enableSignalForSMS();
+        ConversationActions.sendMessage("Sup?");
+        ConversationActions.sendMessage("Sup?");
+        ConversationActions.longPressMessageAt(0);
+        ConversationActions.pressMessageAt(1);
+        ConversationActions.markAsUnread();
+        ConversationActions.goToConversation("+15555123456");
+        Expectations.checkIsShowingDescendantWithId(EspressoUtil.nthChildOf(withId(android.R.id.list), 0), R.id.read_reminder);
+        Expectations.checkIsShowingDescendantWithId(EspressoUtil.nthChildOf(withId(android.R.id.list), 1), R.id.read_reminder);
+    }
+
+    @Test
+    public void checkShouldNotHaveUnreadOptionTest() {
+        ConversationActions.createNewConversation("123456");
+        ConversationActions.enableSignalForSMS();
+        ConversationActions.sendMessage("Sup?");
+        ConversationActions.longPressMessageAt(0);
+        ConversationActions.markAsUnread();
+        ConversationActions.goToConversation("+15555123456");
+        ConversationActions.longPressMessageAt(0);
+        Expectations.checkDoesNotExist(R.id.menu_context_mark_as_unread);
+    }
+
+    @Test
+    public void checkShouldNotHaveUnreadOptionIfMixedTest() {
+        ConversationActions.createNewConversation("123456");
+        ConversationActions.enableSignalForSMS();
+        ConversationActions.sendMessage("Sup?");
+        ConversationActions.sendMessage("Sup?");
+        ConversationActions.longPressMessageAt(0);
+        ConversationActions.markAsUnread();
+        ConversationActions.goToConversation("+15555123456");
+        ConversationActions.longPressMessageAt(0);
+        ConversationActions.pressMessageAt(1);
+        Expectations.checkDoesNotExist(R.id.menu_context_mark_as_unread);
+    }
+
+    // TODO: Check why unread_indicator is not working in the emulator
```

Comment:
will you be keeping this todo? or is it invalidated?

Reply:
I'll change the comment, but essentially the unread_indicator isn't working as expected.